### PR TITLE
krakend 0.4.0 (new formula)

### PR DESCRIPTION
**Pull request by alombarte**
Hello, reopening #23131 as we have reached the 20 needed watchers.

---
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

http://www.krakend.io

Thank you!
-----

### CODE REVIEW

**Comment by alombarte** [comment]
The KrakenD project is in the URL https://github.com/devopsfaith/krakend/ and has all the stars, forks and watchers required. Nevertheless we haven a different repository with the builder to let the users compile the binary themselves is in https://github.com/devopsfaith/krakend-ce/.

Looks like Jenkins is checking the `krakend-ce` repository instead of `krakend`. We created this repo to not distribute binaries (as you prefer compilation over binary placement).

Thanks

**Comment by vinyldarkscratch** [comment]
To whoever reviews this for pulling: install from source and strict audit passed on my local machine as well.  So, the only apparent issue is the noteable repository check, which as @alombarte says, is checking the wrong one.

File: Formula/krakend.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    cd "src/github.com/devopsfaith/krakend-ce" do
+      system "make", "deps"
+      system "make", "build"
+      bin.install "krakend"
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    system "#{bin}/krakend", "--help"
```
Comment (ilovezfs) [inline Formula/krakend.rb]: This needs to be a functionality test

**Review by ilovezfs — CHANGES_REQUESTED** [review]

**Comment by alombarte** [comment]
@ilovezfs I added an assert in the test function to ensure binary is able to run and show the expected version. Nevertheless the previous line you requested to change has been taken from numerous `homebrew-core` formula. As a suggestion maybe this could be included in the website documentation.

Thanks for your time

**Comment by ilovezfs** [comment]
@alombarte a test for functionality means specifically not a `--help` or `--version` test. What you see in old formulae is irrelevant, as we have much higher standards these days for new formulae. For good examples, see `dep`, `faas-cli`, and `jose`.

**Comment by ilovezfs** [comment]
```
  * C: 23: col 1: Use 2 spaces for indentation in a heredoc by using `<<~` instead of `<<~`.
  * C: 38: col 1: Use 2 spaces for indentation in a heredoc by using `<<~` instead of `<<~`.
  * C: 44: col 4: Inconsistent indentation detected.
  * C: 46: col 1: Extra blank line detected.
  * C: 47: col 4: Inconsistent indentation detected.
  * C: 48: col 1: Use 2 spaces for indentation in a heredoc by using `<<~` instead of `<<~`.
```

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
Thanks for your first contribution to Homebrew, @alombarte! Without people like you submitting PRs we couldn't run this project. You rock! :white_check_mark:

**Comment by alombarte** [comment]
Thank you @ilovezfs , but I see the issue closed without merging to master. Is this expected?

**Comment by ilovezfs** [comment]
scroll up.

screen shot 2018-02-02 at 10 28 31 am

**Pull request closed without merging**